src/helpers/files.js

```diff
@@ -43,6 +44,14 @@ const optimalPath = function(from, to) {
 		: to
 }
 
+const getRootPath = function() {
+	if (getCurrentUser()) {
+		return generateRemoteUrl(`dav/files/${getCurrentUser().uid}`)
+	} else {
+		return generateRemoteUrl('webdav').replace('/remote.php', '/public.php')
+	}
+}
+
```

Comment:
You can avoid call `getCurrentUser` twice.

```suggestion
const getRootPath = () => {
    const user = getCurrentUser()
	if (user) {
		return generateRemoteUrl(`dav/files/${user.uid}`)
	}
	
	return generateRemoteUrl('webdav').replace('/remote.php', '/public.php')
}

```

Reply:
`getRootUrl` do not solve your case?

https://github.com/nextcloud/nextcloud-router/blob/8913e9fb9450d4a105a93b306ad297785e1e59fe/lib/index.ts#L190